src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/services/RestrictedPatientCleanup.kt

```diff
@@ -0,0 +1,60 @@
+package uk.gov.justice.digital.hmpps.hmppsrestrictedpatientsapi.services
+
+import com.microsoft.applicationinsights.TelemetryClient
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsrestrictedpatientsapi.gateways.PrisonApiGateway
+import uk.gov.justice.digital.hmpps.hmppsrestrictedpatientsapi.model.entities.RestrictedPatient
+import uk.gov.justice.digital.hmpps.hmppsrestrictedpatientsapi.repositories.RestrictedPatientsRepository
+import javax.persistence.EntityNotFoundException
+import javax.transaction.Transactional
+
+@Service
+class RestrictedPatientCleanup(
+  @Qualifier("PrisonApiGatewayClientCreds") private val prisonApiGateway: PrisonApiGateway,
+  private val restrictedPatientsRepository: RestrictedPatientsRepository,
+  private val telemetryClient: TelemetryClient
+) {
+
+  @Transactional
+  fun deleteRestrictedPatientOnExternalMovementIntoPrison(externalPrisonerMovementMessage: ExternalPrisonerMovementMessage) {
+    if (externalPrisonerMovementMessage.directionCode != "IN") return
+
+    val isValidPrison = prisonApiGateway.getAgencyLocationsByType("INST")
```

Comment:
seems very naff to get all the prisons and then dumping all but one.  Worth calling https://api-dev.prison.service.justice.gov.uk/swagger-ui/index.html#/agencies/getAgency instead

Reply:
Also there is the domain end points for receiving and releasing a prisoner now which are much clearer.  https://offender-events-dev.prison.service.justice.gov.uk/swagger-ui/index.html?configUrl=/v3/api-docs/swagger-config

Reply:
Good shout Mike. I'll listen to receiving events. Will sort in the next PR.